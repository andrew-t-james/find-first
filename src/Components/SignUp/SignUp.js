import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      firstName: '',
      email: '',
      password: ''
    };
  }

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  render() {
    const { firstName, email, password } = this.state;

    return (
      <section>
        <form action="">
          <input
            type="text"
            name="firstName"
            value={firstName}
            onChange={this.handleChange}
          />
          <input
            type="text"
            name="email"
            value={email}
            onChange={this.handleChange}
          />
          <input
            type="password"
            name="password"
            value={password}
            onChange={this.handleChange}
          />
        </form>
      </section>
    );
  }
}

export default SignUp;
