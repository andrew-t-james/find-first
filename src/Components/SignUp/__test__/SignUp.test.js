import React from 'react';
import { shallow } from 'enzyme';
import { SignUp } from '../SignUp';

describe('<SignUp />', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<SignUp />));

  test('should update firstName onChange ', () => {
    const mockEvent = {
      target: {
        name: 'firstName',
        value: 'me'
      }
    };

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('firstName')).toEqual('me');
  });

  test('should update email onChange ', () => {
    const mockEvent = {
      target: {
        name: 'email',
        value: 'me'
      }
    };

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('email')).toEqual('me');
  });

  test('should update password onChange ', () => {
    const mockEvent = {
      target: {
        name: 'password',
        value: 'me'
      }
    };

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('password')).toEqual('me');
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
