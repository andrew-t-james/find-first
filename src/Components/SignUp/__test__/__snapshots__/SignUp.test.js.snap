// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should match snapshot 1`] = `
<section>
  <form
    action=""
  >
    <input
      name="firstName"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      name="email"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </form>
</section>
`;
